include/display.h

```diff
@@ -65,18 +70,27 @@ typedef enum
     ANTIALIAS_RESAMPLE_FETCH_ALWAYS
 } antialias_t;
 
-/** @brief Display context */
-typedef int display_context_t;
+/** 
+ * @brief Display context (DEPRECATED: Use #surface_t instead)
+ * 
+ * @see #surface_t
+ */
+typedef surface_t* display_context_t;
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 void display_init( resolution_t res, bitdepth_t bit, uint32_t num_buffers, gamma_t gamma, antialias_t aa );
-display_context_t display_lock();
-void display_show(display_context_t disp);
+surface_t* display_lock(void);
+void display_show(surface_t* disp);
 void display_close();
 
+uint32_t display_get_width(void);
+uint32_t display_get_height(void);
+uint32_t display_get_bitdepth(void);
+uint32_t display_get_num_buffers(void);
```

Comment:
done

Reply:
Have you committed this yet? It's still in the .c file for me.

Reply:
I reverted it and forgot to mention it. I will do this in a final commit before merging otherwise there are too many conflicts in case other changes are needed

Reply:
Added now